src/rust/rust-proto-clients/src/grpc_client.rs

```diff
@@ -0,0 +1,46 @@
+use std::time::Duration;
+
+use rust_proto::protocol::{
+    endpoint::Endpoint,
+    healthcheck::client::HealthcheckClient,
+    service_client::{
+        ConnectError,
+        Connectable,
+        NamedService,
+    },
+};
+
+use crate::grpc_client_config::{
+    GenericGrpcClientConfig,
+    GrpcClientConfig,
+};
+
+pub async fn get_grpc_client<C: GrpcClientConfig>(
+    client_config: C,
+) -> Result<C::Client, ConnectError> {
+    let GenericGrpcClientConfig {
+        address,
+        healthcheck_polling_interval_ms,
+    } = client_config.into();
+
+    let service_name = C::Client::SERVICE_NAME;
+
+    if !address.starts_with("http") {
+        panic!("Address should start with http, but found: '{address}'")
+    }
+
+    // TODO: Add a `rust-proto` wrapper around tonic Endpoint
+    let endpoint = Endpoint::from_shared(address)?
+        .timeout(Duration::from_secs(10))
+        .concurrency_limit(30);
+
+    HealthcheckClient::wait_until_healthy(
+        endpoint.clone(),
+        service_name,
+        Duration::from_millis(10000),
+        Duration::from_millis(healthcheck_polling_interval_ms),
```

Comment:
all these hardcoded timeouts etc will eventually make their way in to GenericClientConfig